firedrake/mesh.py

```diff
@@ -319,13 +324,15 @@ class MeshTopology(object):
     """A representation of mesh topology."""
 
     @timed_function("CreateMesh")
-    def __init__(self, plex, name, reorder, distribute):
+    def __init__(self, plex, name, reorder, distribute, partitioning=None):
         """Half-initialise a mesh topology.
 
         :arg plex: :class:`DMPlex` representing the mesh topology
         :arg name: name of the mesh
         :arg reorder: whether to reorder the mesh (bool)
         :arg distribute: whether to distribute the mesh to parallel processes
+        :kwarg partitioning: provide a cell partition for running in
+            parallel (useful for debugging, but not necessary).
```

Comment:
This could be merged with `distribute`, no?
- `None`: default behaviour
- `False`: do not distribute mesh
- `True`: distribute mesh (automatically)
- otherwise: distribute mesh according to given partition